hoodie-common/src/test/java/com/uber/hoodie/common/table/view/HoodieTableFileSystemViewTest.java

```diff
@@ -78,6 +79,194 @@ private void refreshFsView(FileStatus[] statuses) {
+    new File(basePath + "/" + partitionPath + "/" + fileName1).createNewFile();
+    new File(basePath + "/" + partitionPath + "/" + fileName2).createNewFile();
+    HoodieActiveTimeline commitTimeline = metaClient.getActiveTimeline();
+    HoodieInstant instant1 = new HoodieInstant(true, HoodieTimeline.COMMIT_ACTION, instantTime1);
+    HoodieInstant deltaInstant2 = new HoodieInstant(true, HoodieTimeline.DELTA_COMMIT_ACTION, deltaInstantTime1);
+    HoodieInstant deltaInstant3 = new HoodieInstant(true, HoodieTimeline.DELTA_COMMIT_ACTION, deltaInstantTime2);
+
+    commitTimeline.saveAsComplete(instant1, Optional.empty());
+    commitTimeline.saveAsComplete(deltaInstant2, Optional.empty());
+    commitTimeline.saveAsComplete(deltaInstant3, Optional.empty());
+
+    refreshFsView(null);
+
+    List<HoodieDataFile> dataFiles = roView.getLatestDataFiles().collect(Collectors.toList());
+    assertTrue("No data file expected", dataFiles.isEmpty());
+    List<FileSlice> fileSliceList = rtView.getLatestFileSlices(partitionPath).collect(Collectors.toList());
+    assertEquals(1, fileSliceList.size());
+    FileSlice fileSlice = fileSliceList.get(0);
+    assertEquals(fileId, fileSlice.getFileId());
+    assertFalse(fileSlice.getDataFile().isPresent());
+    assertEquals(instantTime1, fileSlice.getBaseCommitTime());
+    assertEquals(instantTime1, fileSlice.getBaseInstantForLogAppend());
+    assertFalse(fileSlice.getOutstandingCompactionInstant().isPresent());
+    List<HoodieLogFile> logFiles = fileSlice.getLogFiles().collect(Collectors.toList());
+    assertEquals(2, logFiles.size());
+    assertEquals(fileName2, logFiles.get(0).getFileName());
+    assertEquals(fileName1, logFiles.get(1).getFileName());
+  }
+
+  @Test
+  public void testViewForFileSlicesWithNoBaseFileAndRequestedCompaction() throws Exception {
+    testViewForFileSlicesWithAsyncCompaction(true, false);
+  }
+
+  @Test
+  public void testViewForFileSlicesWithBaseFileAndRequestedCompaction() throws Exception {
+    testViewForFileSlicesWithAsyncCompaction(false, false);
+  }
+
+  @Test
+  public void testViewForFileSlicesWithNoBaseFileAndInflightCompaction() throws Exception {
+    testViewForFileSlicesWithAsyncCompaction(true, true);
+  }
+
+  @Test
+  public void testViewForFileSlicesWithBaseFileAndInflightCompaction() throws Exception {
+    testViewForFileSlicesWithAsyncCompaction(false, true);
+  }
+
+  /**
+   * Helper method to test Views in the presence of concurrent compaction
+   * @param skipCreatingDataFile if set, first File Slice will not have data-file set. This would
+   *                             simulate inserts going directly to log files
+   * @param isCompactionInFlight if set, compaction was inflight (running) when view was tested first time,
+   *                             otherwise compaction was in requested state
+   * @throws Exception
+   */
+  private void testViewForFileSlicesWithAsyncCompaction(boolean skipCreatingDataFile,
```

Comment:
Before every assert, can you add one line as to what that assert is expecting in case the Assert and the variables themselves aren't descriptive enough ?